cmd/xc/commands/transfer.go

```diff
@@ -67,9 +68,14 @@ func CmdTxTransfer() *cobra.Command {
 			}
 			algorithm, _ := cmd.Flags().GetString("algorithm")
 			addressArgs := []xcaddress.AddressOption{}
+			infoArgs := []xclient.GetTxInfoOption{}
 			if algorithm != "" {
 				addressArgs = append(addressArgs, xcaddress.OptionAlgorithm(xc.SignatureType(algorithm)))
 			}
+			if contract != "" {
+				addressArgs = append(addressArgs, xcaddress.OptionContract(xc.ContractAddress(contract)))
```

Comment:
Why is this done?  IMO it's not realistic for treasury to do the same.

People generate the address once, and then it's fixed.  It should be on the `TxBuilder` to use a token address format if needed.